src/main/java/org/gridsuite/cgmes/boundary/server/CgmesBoundaryService.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package org.gridsuite.cgmes.boundary.server;
+
+import org.gridsuite.cgmes.boundary.server.dto.BoundaryInfo;
+import org.gridsuite.cgmes.boundary.server.fullmodel.FullModel;
+import org.gridsuite.cgmes.boundary.server.repositories.BoundaryEntity;
+import org.gridsuite.cgmes.boundary.server.repositories.BoundaryRepository;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.io.UncheckedIOException;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * @author Franck Lecuyer <franck.lecuyer at rte-france.com>
+ */
+@Service
+class CgmesBoundaryService {
+
+    private BoundaryRepository boundaryRepository;
+
+    public CgmesBoundaryService(BoundaryRepository boundaryRepository) {
+        this.boundaryRepository = boundaryRepository;
+    }
+
+    Optional<BoundaryInfo> getBoundary(String boundaryId) {
+        Optional<BoundaryEntity> boundary = boundaryRepository.findById(boundaryId);
+        if (boundary.isPresent()) {
```

Comment:
You can replace this by just a map on the optional

Reply:
Done